### PR TITLE
Use the python-tools action in order to fix pip version

### PR DESCRIPTION
**Pull request by andrewgait**
The pip version currently needs to be fixed to an earlier version to avoid resolver issues (see https://github.com/SpiNNakerManchester/sPyNNaker/issues/1105).

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -43,6 +43,8 @@ jobs:
       - name: Set dynamic environment variables
         run: |
           echo "SPINN_DIRS=$PWD/spinnaker_tools" >> $GITHUB_ENV
+      - name: Install pip, etc
+        uses: ./support/actions/python-tools
```
Comment (dkfellows) [inline .github/workflows/publish.yml:47]: Should also specify:
```
with:
    coverage: false
```
as we don't need support for handling coverage in this build.

**Review by dkfellows — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (dkfellows) [reply .github/workflows/publish.yml:47]: I got that by reading the source at https://github.com/SpiNNakerManchester/SupportScripts/blob/master/actions/python-tools/action.yml

Comment (dkfellows) [reply .github/workflows/publish.yml:47]: Otherwise, looks good; the right sort of fix.

Comment (andrewgait) [reply .github/workflows/publish.yml:47]: Done.

**Pull request merged**